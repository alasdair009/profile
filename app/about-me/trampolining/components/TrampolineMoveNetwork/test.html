<html lang="en">
  <head>
    <style>
      .videos {
        display: flex;
        height: 100vh;
        width: 100vw;
      }
      .videos div {
        align-items: center;
        display: flex;
        flex-direction: column;
        justify-content: center;
        width: 50%;
      }
      iframe {
        aspect-ratio: 16/9;
        width: 90%;
      }
    </style>
  </head>
  <body>
    <div class="videos">
      <div>
        <iframe
          id="standard"
          src=""
          data-src="https://www.youtube.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Standard Embed"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goStandard">Standard</button>
      </div>
      <div>
        <iframe
          id="privacy"
          src=""
          data-src="https://www.youtube-nocookie.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Enhanced privacy"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goPrivacy">Privacy</button>
      </div>
    </div>
    <script>
      const goStandard = document.getElementById("goStandard");
      const standardIFrame = document.getElementById("standard");
      const goPrivacy = document.getElementById("goPrivacy");
      const privacyIFrame = document.getElementById("privacy");
      goStandard.addEventListener(
        "click",
        function () {
          standardIFrame.setAttribute("src", standardIFrame.getAttribute("data-src"));
        },
        false
      );
      goPrivacy.addEventListener(
        "click",
        function () {
          privacyIFrame.setAttribute("src", privacyIFrame.getAttribute("data-src"));
        },
        false
      );
    </script>
  </body>
</html>
