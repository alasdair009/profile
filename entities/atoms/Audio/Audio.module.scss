@use "../../../app/styles/colors";

.root {
  appearance: none;

  &::-webkit-media-controls-panel {
    background: colors.$colorGreenGrass;
  }

  &::-webkit-media-controls-current-time-display,
  &::-webkit-media-controls-time-remaining-display {
    color: colors.$colorWhiteGhost;
  }
}
