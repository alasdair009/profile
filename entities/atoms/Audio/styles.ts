"use client"
import styled from "styled-components";
import {colors} from "@/entities";

export const Root = styled.audio`
    appearance: none;
    
    &::-webkit-media-controls-panel {
        background: ${colors.greenGrass};
    }
    
    &::-webkit-media-controls-current-time-display,
    &::-webkit-media-controls-time-remaining-display {
        color: ${colors.whiteGhost};
    }
`;