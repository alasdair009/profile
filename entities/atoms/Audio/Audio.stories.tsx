import { Meta, StoryObj } from "@storybook/react";
import { Audio } from "./Audio";
import { expect, within } from "@storybook/test";

const meta: Meta<typeof Audio> = {
  component: Audio,
  args: {
    controls: true,
    src: "https://www.w3schools.com/tags/horse.mp3",
    autoPlay: false,
  },
};
export default meta;

export const Default: StoryObj<typeof Audio> = {
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);
    const audioElement = canvas.getByTestId(Audio.name);
    const sourceElement = canvas.getByTestId(`${Audio.name}Source`);

    await expect(audioElement).toBeInTheDocument();
    await expect(sourceElement).toBeInTheDocument();
    await expect(sourceElement).toHaveAttribute("src", args.src);
  },
};
