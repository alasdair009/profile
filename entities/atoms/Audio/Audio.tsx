import { Root } from "./styles";
import { HTMLAttributes } from "react";

type AudioProps = {
  src: string;
  controls?: boolean;
  autoPlay?: boolean;
  loop?: boolean;
} & HTMLAttributes<HTMLAudioElement>;

export function Audio({
  src,
  controls = true,
  autoPlay = true,
    loop = true,
  ...rest
}: AudioProps) {
  return (
    <Root controls={controls} autoPlay={autoPlay} loop={loop} {...rest}>
      <source src={src} type="audio/mpeg" />
    </Root>
  );
}
