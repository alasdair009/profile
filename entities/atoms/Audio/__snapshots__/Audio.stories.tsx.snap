// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/Audio Default play-test 1`] = `
<audio controls
       loop
       data-testid="Audio"
       class="sc-hDcvty flgSKf"
>
  <source src="https://www.w3schools.com/tags/horse.mp3"
          type="audio/mpeg"
          data-testid="AudioSource"
  >
</audio>
`;
